PluginGenerator/DataModel/Rule.cs

```diff
@@ -68,9 +68,42 @@ public XmlCDataSection DescriptionAsCDATA
         [XmlElement(ElementName = "status")]
         public string Status { get; set; }
 
+        [XmlElement(ElementName = "type")]
+        public IssueType Type { get; set; }
+
         [XmlElement(ElementName = "tag")]
         public string[] Tags { get; set; }
 
+        [XmlIgnore]
+        public DebtRemediationFunctionType DebtRemediationFunction { get; set; }
+
+        /// <summary>
+        /// Do not read/write this property directly - use <see cref="DebtRemediationFunction"/> instead.
+        /// </summary>
+        /// <remarks>This property is a hack that allows us to use an enum to specify
+        /// debt remediation function, but to have nothing written to the XML if the
+        /// the value is not specified. There are other ways to do this, but they required
+        /// much more code.</remarks>
+        [XmlElement(ElementName = "debtRemediationFunction")]
+        public string DebtRemediationFunction_DoNotUse_ForSerializationOnly
+        {
+            get
+            {
+                if (DebtRemediationFunction == DebtRemediationFunctionType.Unspecified)
+                {
+                    return null;
+                }
+                return DebtRemediationFunction.ToString();
+            }
+            set
+            {
+                this.DebtRemediationFunction = (DebtRemediationFunctionType)Enum.Parse(typeof(DebtRemediationFunctionType), value);
```

Comment:
Isn't it better to do a `TryParse` and assign `Unspecified` in case of wrong value?

Reply:
I think this is ok; we'll fail fast if the user supplies an invalid value.